src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -101,28 +101,33 @@ public Resource Merge(Resource other)
                 sanitizedKey = attribute.Key;
             }
 
-            object sanitizedValue;
-            if (!IsValidValue(attribute.Value))
-            {
-                OpenTelemetrySdkEventSource.Log.InvalidArgument("Create resource", "attribute value", "Attribute value should be a non-null string, long, bool or double.");
-                sanitizedValue = string.Empty;
-            }
-            else
-            {
-                sanitizedValue = attribute.Value;
-            }
+            object sanitizedValue = SanitizeValue(attribute.Value);
 
             return new KeyValuePair<string, object>(sanitizedKey, sanitizedValue);
         }
 
-        private static bool IsValidValue(object value)
+        private static object SanitizeValue(object value)
         {
-            if (value != null && (value is string || value is bool || value is long || value is double))
+            if (value != null)
             {
-                return true;
+                if (value is string || value is bool || value is long || value is double)
+                {
+                    return value;
+                }
+
+                if (value is int)
```

Comment:
Might want to add `short` here as well. Don't know if there's an easy way to say "convert all numbers to long" but that would be great 😄